src/main/java/org/myrobotlab/service/Runtime.java

```diff
@@ -3865,4 +3852,104 @@ public void addServiceToRequestedInteface(String serviceName) {
     return interfaceToPossibleServices;
   }
 
+  static public Set<String> saveDefault(String className) {
+    try {
+      return saveDefault(className.toLowerCase(), className, null, null);
+    } catch (Exception e) {
+      log.error("saving default config failed", e);
+    }
+    return null;
+  }
+
+  
+  /**
+   * Saves a "sane" set of embedded defaults constructed for this service
+   * @param name - name of service
+   * @param className
+   *          - the class whos defaults will be saved
+   * @return - returns the set of configuration sets successfully saved
+   */
+  static public Set<String> saveDefault(String name, String className) {
+    try {
+      return saveDefault(null, className, null, null);
+    } catch (Exception e) {
+      log.error("saving default config failed", e);
+    }
+    return null;
+  }
+
+  /**
+   * Saves a "sane" set of embedded defaults constructed for this service
+   * 
+   * @param type
+   *          - name of the class with the desired defaults
+   * @param configPrefixPath
+   *          - prefix location to where the configuration sets will be saved
+   * @param overwrite
+   *          - force overwriting existing config sets
+   * @return - set of successfully saved configuration sets
+   * @throws IOException
+   */
+  static public Set<String> saveDefault(String name, String type, String configPrefixPath, Boolean overwrite) throws IOException {
+
+    Set<String> savedPaths = new HashSet<>();
+
+    if (type == null) {
+      log.error("saveDefault className is required");
+      return savedPaths;
+    }
+
+    if (name == null) {
+      name = type.toLowerCase();
+    }
+
+    log.info("saving defaults for {}", type);
+
+    if (overwrite == null) {
+      overwrite = true;
+    }
+
+    if (configPrefixPath == null) {
+      configPrefixPath = "data/config";
+    }
+
+    try {
+
+      Map<String, ServiceConfig> config = ServiceInterface.getDefault(name, type);
+
+      File dir = new File(FileIO.gluePaths(configPrefixPath, name));
+
+      if (dir.exists() && !overwrite) {
+        log.warn("skipping %s - directory already exists", name);
+        return savedPaths;
+        // continue;
+      }
+
+      dir.mkdirs();
+
+      for (String service : config.keySet()) {
+        String path = FileIO.gluePaths(dir.getAbsolutePath(), service + ".yml");
+        FileIO.toFile(path, CodecUtils.toYaml(config.get(service)));
+      }
+
+      if (!config.containsKey("runtime")) {
+        // config did not come with an explicit runtime
+        // therefore we will create one with the order of the keyset
+        Map<String, ServiceConfig> runtimeConfig = ServiceInterface.getDefault("runtime", "Runtime");

```

Comment:
to be explicit about the ordering of the keyset here, you probably should make this a LinkedHashMap